main.go

```diff
@@ -0,0 +1,260 @@
+			"Number of apps.",
+			nil,
+			nil,
+		),
+		appQueue: prometheus.NewDesc(
+			prometheus.BuildFQName(namespace, "", "app_queue"),
+			"Number of requests in app process queues.",
+			[]string{"name"},
+			nil,
+		),
+		appProcsSpawning: prometheus.NewDesc(
+			prometheus.BuildFQName(namespace, "", "app_procs_spawning"),
+			"Number of processes spawning.",
+			[]string{"name"},
+			nil,
+		),
+		requestsProcessed: prometheus.NewDesc(
+			prometheus.BuildFQName(namespace, "", "requests_processed"),
+			"Number of processes served by a process.",
+			[]string{"name", "pid"},
+			nil,
+		),
+		procUptime: prometheus.NewDesc(
+			prometheus.BuildFQName(namespace, "", "proc_uptime"),
+			"Number of seconds since processor started.",
+			[]string{"name", "pid"},
+			nil,
+		),
+		procMemory: prometheus.NewDesc(
+			prometheus.BuildFQName(namespace, "", "proc_memory"),
+			"Memory consumed by a process",
+			[]string{"name", "pid"},
+			nil,
+		),
+		procStatus: prometheus.NewDesc(
+			prometheus.BuildFQName(namespace, "", "proc_status"),
+			"Running status for a process.",
+			[]string{"name", "pid", "codeRevision", "lifeStatus", "enabled"},
+			nil,
+		),
+	}, nil
+}
+
+// Collect fetches the statistics from the configured memcached server, and
+// delivers them as Prometheus metrics. It implements prometheus.Collector.
+func (e *Exporter) Collect(ch chan<- prometheus.Metric) {
+	info, err := e.status()
+	if err != nil {
+		ch <- prometheus.MustNewConstMetric(e.up, prometheus.GaugeValue, 0)
+		log.Errorf("failed to collect status from passenger: %s", err)
+		return
+	}
+	ch <- prometheus.MustNewConstMetric(e.up, prometheus.GaugeValue, 1)
+
+	// TODO: Doesn't seem to be a queryable uptime on the main process.
+	// Talk with @grobie about how to handle this.
+
+	ch <- prometheus.MustNewConstMetric(e.version, prometheus.GaugeValue, 1, info.PassengerVersion)
+
+	// This value should be 0. Not sure if I should use a gauge or a
+	// counter, as we shouldn't be counting anything.
+	ch <- prometheus.MustNewConstMetric(e.toplevelQueue, prometheus.CounterValue, parseFloat(info.TopLevelRequestsInQueue))
+	ch <- prometheus.MustNewConstMetric(e.maxProcessCount, prometheus.GaugeValue, parseFloat(info.MaxProcessCount))
+	ch <- prometheus.MustNewConstMetric(e.currentProcessCount, prometheus.GaugeValue, parseFloat(info.CurrentProcessCount))
+	ch <- prometheus.MustNewConstMetric(e.appCount, prometheus.GaugeValue, parseFloat(info.AppCount))
+
+	for _, sg := range info.SuperGroups {
+		ch <- prometheus.MustNewConstMetric(e.appQueue, prometheus.GaugeValue, parseFloat(sg.RequestsInQueue), sg.Name)
+		ch <- prometheus.MustNewConstMetric(e.appProcsSpawning, prometheus.GaugeValue, parseFloat(sg.RequestsInQueue), sg.Group.ProcessesSpawning, sg.Name)
+
+		for _, proc := range sg.Group.Processes {
+			ch <- prometheus.MustNewConstMetric(e.procMemory, prometheus.GaugeValue, parseFloat(proc.RealMemory), sg.Name, proc.PID)
```

Comment:
You mention that using PID will probably create too many metrics. While this is * probably * true (there are no guarantees on the lifetime of a single process, it depends on load, and it seems like some configs actually lock a min_proc value so that they'll never have fewer than the max number desired), the process churn will be user specific.

The bigger question is, are these metrics desired? They are process-level metrics, so I don't know how to log them other than with their PID to make them unique. Unless we can send multiple times to the channel and note the app name but not make them differentiable by pid? I think some of these could be interesting, e.g. requestsProcessed.